src/gax-internal/src/observability/errors.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use gax::error::rpc::Code;
+use http::StatusCode;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ErrorType {
+    HttpError {
+        code: StatusCode,
+        reason: Option<String>,
+    },
+    ClientTimeout,
+    ClientConnectionError,
+    ClientRequestError,
+    ClientRequestBodyError,
+    ClientResponseDecodeError,
+    ClientRedirectError,
+    Internal,
+}
+
+// Trait to abstract reqwest::Error for testing
+pub trait ReqwestErrorDetails {
+    fn is_timeout(&self) -> bool;
+    fn is_connect(&self) -> bool;
+    fn is_request(&self) -> bool;
+    fn is_body(&self) -> bool;
+    fn is_decode(&self) -> bool;
+    fn is_redirect(&self) -> bool;
+}
+
+impl ReqwestErrorDetails for reqwest::Error {
+    fn is_timeout(&self) -> bool {
+        self.is_timeout()
+    }
+    fn is_connect(&self) -> bool {
+        self.is_connect()
+    }
+    fn is_request(&self) -> bool {
+        self.is_request()
+    }
+    fn is_body(&self) -> bool {
+        self.is_body()
+    }
+    fn is_decode(&self) -> bool {
+        self.is_decode()
+    }
+    fn is_redirect(&self) -> bool {
+        self.is_redirect()
+    }
+}
+
+impl ErrorType {
+    pub(crate) fn from_reqwest_error(err: &dyn ReqwestErrorDetails) -> Self {
+        if err.is_timeout() {
+            ErrorType::ClientTimeout
+        } else if err.is_connect() {
+            ErrorType::ClientConnectionError
```

Comment:
optional: I think I would have used a match:


```suggestion
       match err {
         e if e.is_timeout() => ErrorType::ClientTimeout.
         e if e.is_connect() => ErrorType::ClientConnectionError.
         ... ...
         _ => Error::Internal
```